src/expr/src/expr/expr_binary_nullable.rs

```diff
@@ -78,6 +80,43 @@ pub fn new_nullable_binary_expr(
     }
 }
 
+fn build_array_access_expr(
+    ret: DataType,
+    l: BoxedExpression,
+    r: BoxedExpression,
+) -> BoxedExpression {
+    macro_rules! array_access_expression {
+        ($array:ty) => {
+            Box::new(
+                BinaryNullableExpression::<ListArray, I32Array, $array, _>::new(
+                    l,
+                    r,
+                    ret,
+                    array_access,
+                ),
+            )
+        };
+    }
+
+    match ret {
+        DataType::Boolean => array_access_expression!(BoolArray),
+        DataType::Int16 => array_access_expression!(I16Array),
+        DataType::Int32 => array_access_expression!(I32Array),
+        DataType::Int64 => array_access_expression!(I64Array),
+        DataType::Float32 => array_access_expression!(F32Array),
+        DataType::Float64 => array_access_expression!(F64Array),
+        DataType::Decimal => array_access_expression!(DecimalArray),
+        DataType::Date => array_access_expression!(NaiveDateArray),
+        DataType::Varchar => array_access_expression!(Utf8Array),
+        DataType::Time => array_access_expression!(NaiveTimeArray),
+        DataType::Timestamp => array_access_expression!(NaiveDateTimeArray),
+        DataType::Timestampz => array_access_expression!(PrimitiveArray::<i64>),
+        DataType::Interval => array_access_expression!(IntervalArray),
+        DataType::Struct { .. } => array_access_expression!(StructArray),
+        DataType::List { .. } => array_access_expression!(Utf8Array),
```

Comment:
```suggestion
        DataType::List { .. } => todo!()
```

Is returning varchar correct? We can leave it a todo here if we haven't figured out how to do it.

Reply:
You're right that's a mistake + returning a List is not implemented yet so it should be a todo!() here

Reply:
Because of this mistake, one of the test passed which should not have passed. I rewrote that test now as well.